### ISSUE TITLE
Windows.Gaming.UI.Gamebar.Visible: Samples would be appreciated

### ISSUE THREAD

**Issue by SaadBazaz** [MicrosoftDocs/winrt-api#1944]
Hi,
I've been trying to call Windows.Gaming.UI.Gamebar.Visible on a WPF app (targeting .NET Framework 4.6.1) using the _Microsoft.Windows.SDK.Contracts_ . However, calling

```
        if (Windows.Gaming.UI.Gamebar.Visible)
```

Always returns false. Even when the Gamebar is open!

This makes me think that I might be doing something wrong. I'd appreciate some more documentation on this and why / how it might NOT give the appropriate value.

Link to documentation: https://github.com/MicrosoftDocs/winrt-api/blob/docs/windows.gaming.ui/gamebar_visible.md

**Comment by stevewhims** [comment]
Hi, @SaadBazaz, and thanks for filing this issue. I've reached out to the feature team to see whether there's an opportunity to improve the documentation here.

Incidentally, feel free to visit an actual published topic (for example, https://docs.microsoft.com/uwp/api/windows.gaming.ui.gamebar.visible), scroll down to the Feedback section, and click Submit and view feedback for > This product (if you have feedback about the API itself), or This page (if you have feedback about the documentation itself). That'll get routed to the right place more quickly than filing against the repo.

Thanks!
Steve

**Comment by stevewhims** [comment]
UPDATE: The feature team are discussing this internally. I just pinged the thread again to see what the official answer is.

Thanks for your patience.

-Steve

**Comment by SaadBazaz** [comment]
Hi @stevewhims ,
Thank you for the kind answer. I apologize for the lack of participation from my side, life had me there.
I really appreciate that the team is looking into it. The project I'm doing depends on this feature so it'd be a great help, as there's not much help online relevant to the Gamebar.

I had visited the link you'd shared earlier but it reflected the same written here. I'll keep in mind your recommendations from now on.

**Comment by SaadBazaz** [comment]
Hey @stevewhims ,
Hope you're well. Was wondering if there was any update on this. Thanks again for your cooperation.

**Comment by stevewhims** [comment]
Hi, @SaadBazaz, my apologies for the delay. The feature team have said that "The Game Mode APIs are deprecated in Windows 10, version 1809 and later" (see https://docs.microsoft.com/previous-versions/windows/desktop/gamemode/game-mode-portal). And it sounded like they wanted to mark **Windows.Gaming.UI.Gamebar.Visible** as deprecated, also. We didn't yet get to the full list of deprecated APIs in this area, but it does seem like this particular one is deprecated, and shouldn't be used. I know that's not the answer you were hoping for.

I'll go ahead and close this issue now, but I'll be be happy to reopen it and/or add deprecation notices to the affected APIs just as soon as that info comes in. I've let the feature team know I'm awaiting that info.

Many thanks!
Steve

**Comment by SaadBazaz** [comment]
Yeah that'd be great. I really need to use a "Gamebar detection" concept in the project I mentioned. I'd appreciate if they could also link us up with the upgraded APIs.

Since the Gamebar SDK only just got released, it's a very lucrative area in which I believe there should be more attention.

Cheers

**Comment by zohairhadi** [comment]
I was also facing this issue, I wanted to know that has it been fixed?

**Comment by SaadBazaz** [comment]
@stevewhims Hey there, is there any update on this Issue?

**Comment by stevewhims** [comment]
Hi, all! So, all the info that I have to date is in this thread. I have just pinged the feature team again to let them know that there's still interest in this thread. I'll be be happy to reopen it and/or add deprecation notices to the affected APIs just as soon as I get that info.

-Steve

**Comment by stevewhims** [comment]
For the feature team, would you be able to explain why you want to call this API (that is, Windows.Gaming.UI.Gamebar.Visible)? The feature team's thought here is that it doesn't have any value to developers, so we want to confirm we're considering the right problem.

Many thanks!
-Steve

**Comment by SaadBazaz** [comment]
Hey Steve,
So basically I want to create a simple app which can launch / close the Xbox gamebar by the press of a special controller button. And when the gamebar is open, using the directional buttons on the controller I'd like to be able to navigate through the Gamebar.

The controller in question: Xbox 360 Controller.

Unfortunately the gamebar only works for the Xbox One controller which is quite unfortunate considering they just had to map a few buttons of the 360 controller (Plus, it has the same buttons!)

So I decided to tackle that issue on my own.

For that, I want to map the 360 controller's keys to the keyboard keys *only when the gamebar is visible*. It's a quick hack which does the job pretty well. That's the reason why I would have liked to have this API. This way I can switch modes when Gamebar is active (controller->keyboard) vs when Gamebar is invisible (back to controller).

Sorry for the long comment but that's as much as it goes. If the feature team can instead just map the 360 keys to the gamebar then that'll save a lot of other work but that's their choice I suppose 😄

**Comment by stevewhims** [comment]
Thank you, @SaadBazaz! I've relayed that info back to the feature team (and I've reopened this issue now that we're seeing traction).

Thanks for your patience!

-Steve

**Comment by SaadBazaz** [comment]
Thanks Steve! Looking forward to seeing some build-up here. It's exciting! 🤩

**Comment by stevewhims** [comment]
Hi, @SaadBazaz, some more questions from the feature team, if you don't mind.

"In testing this, the 360 controller appears to work almost identically to the Xbox One controller. Is it possible that you (the user) used some 3rd party software to remap buttons, and now it's not being interpreted correctly?"

"Also, it's not clear what you (the user) are trying to do. You want an app that mimics button presses on a controller? I’m not sure what the use case for this is. Or you're creating an app because your 360 controller doesn’t work (when this should actually be supported)?"

Thanks!
Steve

**Comment by SaadBazaz** [comment]
Hey Steve, 
Thanks for the response and the wait!

## My Setup
I've been using the official Microsoft Driver (mirror link [here](https://www.driverguide.com/driver/detail.php?driverid=2004462), it seems like the official Microsoft website doesn't host it anymore). The controller in question is an original Microsoft Xbox 360 Wireless Controller. I'm using a third-party wireless receiver. 

It works perfectly, the driver works immediately with the receiver hardware, and the computer detects the Controller.

## Some things to clarify
- I do not use remapping software. 
- Pressing the "Guide" button on the controller does nothing. It doesn't even turn off the Controller if I long-press!
- Multiple [forums](https://gaming.stackexchange.com/questions/116670/how-can-i-turn-off-a-wireless-xbox-360-controller-for-windows) also report that the "Guide" button can neither open the Gamebar (single click), nor can it close the controller (hold). They refer to third-party software which do the simple mapping of (pseudocode):
```
if pressedGuideButton:
    XInput.Close_Controller_API ( )
```

## What I would like to do
I would simply like to map the Guide button to close the controller on **long-hold**, and open the Gamebar on **tap**. I did that successfully by mapping ```if long-hold: Xinput.Close_Controller_API ( )``` and ```if tap: KeyboardSimulator ("WIN" + "G")```. The next step is to **control the Gamebar** (Move the highlights and select options / go back).

## Possible Solutions to Control the Gamebar
- Source-code level driver implementation.
**FAIL.** Reason: I do not have access to Windows Source code, nor do I know how to make device drivers.

- Map Xbox 360 controller to emulate Xbox One Controller.
**FAIL.** Reason: Cannot find Xbox One Controller API in Windows so I can emulate One's Guide button.

- Map Xbox 360 controller to emulate keyboard presses but **only when the Gamebar is visible**.
**SUCCESS WITH ISSUES.** Issue: I need to know when the Gamebar is visible, so that I emulate the keyboard ONLY THEN!

# TLDR;
_Answer to Q1:_ No.
_Answer to Q2:_ Creating this app because this is a feature which should be implemented in Windows, but it isn't.

**Comment by stevewhims** [comment]
Thanks, @SaadBazaz, for the detailed answer. I've relayed that info back to the feature team (sorry for my delay this time).

Thanks again!
Steve

**Comment by stevewhims** [comment]
Thanks, @SaadBazaz. Here's what two product team members say:

"None of the coding listed in this app should be necessary, given that the controller should work with Game Bar. We've tested with wired controllers (but not wireless 360 controllers). You (the customer) call out that you're using a 3rd party wireless receiver. I wonder whether that's not correctly sending the Nexus/Guide button key press through. In that case that would be an issue with the 3rd party product. You also mention that long press doesn't work to turn off the controller. Again, that could be a problem with the 3rd party receiver, or possibly that the nexus button just doesn’t work. In any case, we recommend using a wired 360 controller; perhaps the 360 controller that you have can be connected with a wire for testing."

"We test Game Bar with a wired 360 controller. We just tried it again and everything is working fine from what we can tell on the latest generally available release of the Windows client OS (Windows 10, version 2004 (10.0; Build 19041))."

It seems safe to say that there's no action required here to update the documentation, which is how this issue originated ("I'd appreciate some more documentation on this").

I'll leave the issue open a little longer to give you a chance to respond, and if you'd like some links to support resources, then I can provide those.

Many thanks!
Steve

**Comment by SaadBazaz** [comment]
Hey Steve,
Thanks for the response. I'd like to respond to some things pointed out by the team.

1. I tested two **original** controllers, both had this same issue (Nexus/Guide button doesn't turn off the controller). I believe that since I'm able to turn them **on** using the Nexus/Guide button, and that similar apps made by other people work with my Nexus/Guide (the controller turns off successfully!), then there is nothing wrong with my buttons.
2. The third-party receiver part should not be true, as far as I know. Because the **driver** which is used alongside it, is the original one. So it seems like more of an issue in the **driver**. Plus, as mentioned before, apps made by other people which turns of the controller, work.
3. The issue is in the 360 wireless, not the wired to begin with. So I don't think it's a possible comparison.
4. There are quite a lot of complaints on forums, Microsoft and non-Microsoft alike, which mention this issue. It affects a lot of people, not just me.
Some forums mentioning the issue. They say that it doesn't work for original receivers as well:
- https://www.sevenforums.com/gaming/198746-how-turn-off-360-wireless-controller-pc.html
- https://gaming.stackexchange.com/questions/116670/how-can-i-turn-off-a-wireless-xbox-360-controller-for-windows

Some apps which people have already built, which solve this issue:
- https://pety.me/XboxExt/
- https://github.com/mina-asham/XBox360ControllerManager

So I don't think I am alone on this. It's just that I saw an opportunity of getting it to work with the Gamebar, that's all. For the problem of turning off the controller, the interwebs have already blessed us with solutions.

However, even then if I'm missing out on something I really need to install to get this work, or if you even want me to go out and buy an original receiver, please do let me know and share any resources, links or anything. I find it fun to explore new venues and fix problems.

**Comment by stevewhims** [comment]
Thanks, @SaadBazaz. I've passed that info along to the feature team.

Since the feature team is now aware of the forum threads that you mentioned, I feel like those are better places to continue this discussion than here, for several reasons. Mainly that this feedback channel is for feedback about the documentation, and I think we've established that this thread won't lead to any documentation changes in the near term (if there are any, I'll hear about them from the feature team). And secondly there'll be a direct link on forum threads (or other support channels) to the feature team, rather than going through an intermediary, as has been happening here (with all the delays and extra overhead and potential miscommunications that that can involve).

So I'll close this issue now. But here are all of the developer support resources I'm aware of, just in case you need them.

- [Microsoft Q&A](https://docs.microsoft.com/answers/topics/windows-uwp.html) is the right place to lobby for a change or a new feature, to make a suggestion, or to ask a product question.
- For community assistance with development questions, there are our [forums](https://social.msdn.microsoft.com/Forums/newthread?category=windowsapps&forum=wpdevelop&prof=required), and [StackOverflow](https://stackoverflow.com/questions).
- The [Windows developer support](https://developer.microsoft.com/windows/support) site explains other support options.
- The [Visual Studio developer community](https://developercommunity.visualstudio.com) is available, and responsive. The former User Voice feature is now there on the **Suggest a feature** tab, if you want to lobby for a change, a feature, or to make a suggestion.
- For developer issues, or suspected product bugs, there's [Microsoft Support](https://support.microsoft.com/contactus); a support engineer will look at the repro and see whether the issue is a Microsoft product bug, or something else. If it's a bug with a Microsoft product then the ticket will be free of charge.

-Steve

**Comment by SaadBazaz** [comment]
Thanks a lot, @stevewhims ! It really means a lot that you looked into this even though I made mistakes, and for guiding me to the right place. I'll check out where I can push this change further.

**Comment by stevewhims** [comment]
Thank you, @SaadBazaz. :)
